### ISSUE TITLE
google_privateca_certificate_authority data source should not attempt to fetch the CSR for subordinate CAs that aren't in AWAITING_USER_ACTIVATION state

### ISSUE THREAD

**Issue by cameronhimself** [hashicorp/terraform-provider-google#12432]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

Terraform v1.2.8

### Affected Resource(s)

* `google_privateca_certificate_authority` data source

### Terraform Configuration Files

```tf
data "google_privateca_certificate_authority" "default" {
  location = "us-east1"
  pool = "pool"
  certificate_authority_id = "subordinate-ca"
}
```

### Expected Behavior

The data source should be retrieved with `pem_csr` being null.

### Actual Behavior

`Error when reading or editing PrivatecaCertificateAuthority "projects/{project}/locations/{location}/caPools/{pool}/certificateAuthorities/{ca}": googleapi: Error 400: Cannot perform Certificate Authority CSR retrieval, Certificate Authority is in state ENABLED.`

### Steps to Reproduce

1. Create a root CA and subordinate CA.
2. Activate and enable the subordinate CA.
3. Attempt to pull the subordinate CA in as a data source

**Comment by cameronhimself** [comment]
Addressed by #12435.

**Comment by cameronhimself** [comment]
Aaaand addressed upstream by https://github.com/GoogleCloudPlatform/magic-modules/pull/6496
